XP 2.0: Cauldron Re-work

MonsieurWTF opened the pull request:
This contains all of the changes to just the Cauldrons, and nothing else regarding the current Enchanting/Repair factories on the server.

MonsieurWTF commented:
Run wild @ProgrammerDan

ProgrammerDan commented on config.yml:
```diff
@@ -7723,3 +7441,351 @@ production_recipes:
       Sea Lantern:
         material: SEA_LANTERN
         amount: 16
+          Wood_Plains_Crate:
```
Too many spaces here.

ProgrammerDan commented on config.yml:
```diff
@@ -7723,3 +7441,351 @@ production_recipes:
+      Baked Potato:
+        material: BAKED_POTATO
+        amount: 512
+      Awkward Potion:
```
This seems wrong; durability: EDIT?

MonsieurWTF commented:
Still figuring out what durability Awkward Potion is in the github. Dura 1?

ProgrammerDan commented:
Outside my depth, although I recall finding a comprehensive listing at one point, naturally can't remember where/how to find it. 

Note you can use the "Add a line note" function to reply to a specific comment against the diff.

Generally clean, although since you relocated some of the factory definitions within the file this is going to cause merge problems. I'd recommend keeping them where they were, rather then moving them to make things neater. Trust me on this :D

MonsieurWTF commented:
Figured it out, Durability 16 according to the game.

MonsieurWTF commented:
Should I put the Emerald cauldron at the bottom of the list, or since it's new it'll be fine wherever?

ProgrammerDan commented:
Since it's new, it should be safe anywhere *in theory*, I recommend putting it at the end of the largest block of factory definitions (before the "all recipe" section starts).

MonsieurWTF commented:
Alright, adjusted most of that, I believe. Hopefully not too many mistakes remain.

@ttk2 Still have questions about the cauldron changes? I can post more about it here tomorrow, but I've probably gotta call it here soon now. At the least I have the files available at any time to be tested.

ProgrammerDan commented:
Dunno your level of gitfu, but depending on how you want to approach these fixes, you could make the changes and use rebase locally to "squash" your commit into a single one against your master. That'll help keep master cleaner if these changes do get merged, and hides all the fixin's you've made locally (and makes future rebases less likely to fail!)

MonsieurWTF commented:
gitfu status = newbie.

ProgrammerDan commented:
ok, I'll doublecheck my gitfu on the command for you and test it locally, then recommend it here :).

MonsieurWTF commented:
Alright, coolio. Hope I'm doing decent updatin' this stuff fast and keepin' it clean. That'll be all for me today.

ProgrammerDan commented:
Pretty impressed tbh. Have a great night.

ttk2 commented:
I can't make any right away suggestions, need a summary and then the fun of making a decision.

MonsieurWTF commented:
Summary of changes:

- Wood Cauldron reworked:
 - Now has 4 simple recipes and 4 Crate recipes
 - 4 simple recipes have the standard 1280 ingredient + Bottles -> XP
 - 4 Crate recipes are designed for 'local industry' use and promote the resource gathering in 1 of 4 biomes to make the large quantities of raw materials reasonable. (Produces 1 Wood Crate)
- Iron Cauldron reworked:
 - Price change increase (added 5 Emerald block cost)
 - Now has 3 simple recipes and 1 Crate recipe (Smaragdus recipe was not changed)
 - 3 simple recipes have the variations of various biome ingredients.
 - Crate recipe requires 4 Wood Crates, plus a variety of global materials. (Produces 1 Iron Crate)
- Diamond Cauldron reworked:
 - Price change increase (added 40 Emerald blocks, 200 Iron ingots, 50 ->160 diamonds)
 - Now has 1 Crate recipe and 1 Conversion recipe
 - Converts Iron Crate into 15 Emeralds
 - Crate recipe requires 2 Iron Crates and a larger assortment of global ingredients. (Produces 1 Diamond Crate)
- Emerald Cauldron from #86 reworked:
 - Now has 1 Crate recipe and 1 Conversion recipe
 - Converts Diamond Crate into 6 Emerald Blocks and 4 Emeralds.
 - Crate recipe requires 2 Diamond Crates and multiple luxury goods of varying cost (Produces 29 Emerald Blocks)

MonsieurWTF commented:
Point of order: When I say 'Crate', I mean the output, lore-based item of a Wood/Iron/Diamond Crate containing a variety of 48+ stacks of materials. It does not rely on compacting the individual ingredients to produce XP.

MonsieurWTF commented on config.yml:
```diff
@@ -7723,3 +7442,370 @@ production_recipes:
+        material: ENDER_CHEST
+        amount: 1
+        display_name: Emerald Crate
+        lore: The finest luxury goods on the market packaged into one crate.
```
Somehow put this back in, I'll fix that then rebase this...however I do that. I'll figure it out tonight.

ProgrammerDan commented:
Rebase tips: http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html

I fell asleep before writing this up, sorry man. Not sure if the above link is clear, if it isn't, I'll try to do it locally and report back the steps.

MonsieurWTF commented:
Kk, needed to lookup some more stuff, but seems the rebase worked correctly.

MonsieurWTF commented:
Btw, how did the numbers look on your iron doc?

ProgrammerDan commented:
Finally home for the evening so that's next on my list. Will do matta's and
yours.
On Jun 11, 2015 8:21 PM, "MonsieurWTF" <notifications@github.com> wrote:

> Btw, how did the numbers look on your iron doc?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Civcraft/FactoryMod/pull/91#issuecomment-111313089>.
>

MonsieurWTF commented:
Gotcha

MonsieurWTF commented:
So uhhh....how're the prices?

MonsieurWTF commented:
Prices have been set, now I just need to go back in to this and edit, commit, and squash tonight.

ProgrammerDan commented:
Just a reminder, don't squash any further commits just yet; simply push the new commits to your repository.

MonsieurWTF commented:
Values and Emerald outputs have been updated. Thoughts on changes?

I tried to get them to line up value-wise with your suggestions on changes, making sure to adjust it percentage-wise so as not to accidentally overdo it.

- Overall, the Wood Crates should be ~55 ingots for each.

- Iron Crate should go up by ~35 ingots in value to 85-90 value-added.

- Diamond Crate no longer needs Compact Ice and just needs regular Ice.

- Emerald Crate gained a few extra lava buckets.

- Each level of crate gained more emeralds to counteract the cost increase of Wood and Iron Crate adjustments.

ttk2 commented:
some people are really opposed to Lava, but most I talked to said it was ok because it only required a little per large amount of XP, increases will not be well received, although if you can compact it the game may change a lot.

MonsieurWTF commented:
The lava buckets were purposely left at that small amount when there was enough Emerald blocks being produced so as to avoid a situation that would cause Lava to be consumed at an incredibly-rapid pace, but in my opinion I think the only group of people that would be proposed to its inclusion would be players that either don't like non-renewables for high-value returns, or don't like materials that can't be easily botted/mass-produced.

if it were possible I'd spend some more time trying to figure out the best way to add compact-able ingredients into the Diamond and Emerald tiers, but I don't want to have some items have massive quantities at that stage, while the others remain at 1 stack or less. If anyone has suggestions as to what could be added that would make sense there, or at the Iron cauldron level, feel free to weigh in.

ProgrammerDan commented on config.yml:
```diff
@@ -7723,3 +7441,355 @@ production_recipes:
+        amount: 640
+      Glass Bottle:
+        material: GLASS_BOTTLE
+        durability: 16
```
What is a durability 16 glass bottle?

MonsieurWTF replied:
My dumb ass forgetting to remove durability and trying to edit the amount.

I'll get that fixed up tonight. Should I be planning to squash or no squash still?

ProgrammerDan replied:
No squash yet.

ProgrammerDan commented on config.yml:
```diff
@@ -7723,3 +7441,355 @@ production_recipes:
+      Pumpkin Pie:
+        material: PUMPKIN_PIE
+        amount: 64
+      Compact Ice:
```
Might want to change the title to "Ice"

ProgrammerDan commented on config.yml:
```diff
@@ -7723,3 +7441,355 @@ production_recipes:
+        amount: 2
+        display_name: Iron Crate
+        lore: Processed market supplies. It can be converted into XP but it's not very efficient...
+      Golden Carrot:
```
You mentioned you wanted this to express ores in the recipe. Until the recent pigman apocalypse/shortage, gold was never expressed from ore unless the ore itself was in the recipe.

MonsieurWTF replied:
Is the gold here still okay, or no? With the sudden shrink in production capability (Although TTK patched something this morning that fixed it for a few areas) there might be some people converting the gold ores into ingots (which some have).

ProgrammerDan replied:
Well, it's 64 ingots total, or 128 per E cauldron run -- that's not terrible, and added demand on the gold market will have some interesting effects. They are just super grindy to make.

MonsieurWTF replied:
golden carrots take nuggets though?

MonsieurWTF replied:
I think it'd be interesting to see a more serious use out of gold than just the standard tools or pyramid bases that people only use them for now. It is grindy, yes, but people use the portal farms already for other competitive goods, so I think that by adding in a 3rd opportunity for the nuggets to be utilized, we'll see players potentially even starting to use the gold ore smelting more than previous.

ProgrammerDan replied:
It'd be an interesting market balance test. Gold Ore is rarer then Redstone ore in a casual survey of what I've found where; it's more rare than lapis. Yet, it has less market value. Gold in general is depressed in value because of the ready alternatives to ore from simply building better gold farms. I'm not sure how it would balance out!

ProgrammerDan commented on config.yml:
```diff
@@ -7723,3 +7441,355 @@ production_recipes:
+        material: OBSIDIAN
+        amount: 384
+      Eye of Ender:
+        material: EYE_OF_ENDER
```
Meant to mention, this requires Aspect Elementizer; I kind of like the post-end-of-life use for this factory, but I'm also fairly certain all but Carson let theirs cull. Not sure that matters, just thought I'd mention.

MonsieurWTF replied:
Yup, I found that out yesterday after reviewing the recipe for eye of ender and realized that this works out pretty well. Yes, you are correct, the Aspect Elementizer has no real use use after it is constructed (a shame, really. No factory should be useless after a single purpose) aside for making the Eyes of Ender. I imagine with that ingredient in the Emerald Cauldron recipe, it'll seriously bring to question what groups of players can actually afford to go all the way to the top, and what groups have to take charge in cooperating to get their input quotas reached.

MonsieurWTF commented:
Things changed in latest commit:

- More Price Balancing, as was outlined by Dan, along with editing improper names and durabilities/amounts
- Changed output of Emerald Crate to output all 16 Buckets used to store the lava. (~30 ingot decrease in final price)
- Adjusted production time of Iron XP solo-recipes from 10 -> 20

MonsieurWTF commented:
Hmm, more interesting pushback today that I was expecting yesterday, regarding the consumption of Dirt and Lava for XP production. Any opinions on it @ProgrammerDan ? I've been considering drawing back the Lava a tad bit since its cost in the final Cauldron doesn't have a significant impact and seems to only be there for diversity and its non-renewable property.

I'm considering lowering it to either 12, 8, or 4.

ProgrammerDan commented:
Well, lava was one of the more interesting ingredients. From a balance perspective it doesn't matter much. The dirt is also not a big deal, cost wise.

Perhaps propose changing it to Cobblestone instead of sand. In terms of "cost" on market, they are identical.

MonsieurWTF commented:
Did you mean Dirt instead of Sand? The only real quantity of sand being consumed in the Crates is for the Desert Crate, and for the bottles at each Conversion level.

Should I replace the Glass Bottles with Cobblestone to take those off of the non-renewable list?

ProgrammerDan commented:
Sorry, I mean, replace all "Dirt" with "Cobblestone" in the recipes. Leave your Sand and bottle requirements alone. Lack of sleep is causing some mental confusion.

MonsieurWTF commented:
Gotcha.

Still though...this is dirt we're talking about. Cobblestone can be so easily botted you don't even need a macromod for it, you can set up your own cobble grinder using water and lava and just go at it.

Not to mention there's shittons of it laying everywhere because people always go mining, so that overall change is just something to make it even easier for players.

ProgrammerDan commented:
I know, I know. What about gravel? Cut the numbers in half and use gravel instead. It spawns about as often in the map as dirt but almost exclusively underground, and its only current use is lava fill.

MonsieurWTF commented:
That sounds good. I could even split it at 1 Stack dirt, and the Gravel is 50% of what dirt was supposed to be, so 3 stacks Gravel per Wood Crate.

That would come down to 48 stacks Gravel per run and 16 stacks Dirt.

Maybe a little too high. I'll take off 16 stacks to account for the dirt.

Sooo 32 stacks of Gravel, 16 stacks of Dirt?

ProgrammerDan commented:
Just remove the dirt component entirely. I get gravel at a 1 to 2 rate in comparison to dirt while mining, so just straight replace dirt with gravel at 50% of volume to maintain balance.

MonsieurWTF commented:
Okay, 3 stacks per run, 48 stacks per Emerald Crate. I'll get that swapped in later today.

To get back to the lava, if there's about 16 stacks of blocks made daily, then that'll be 480 blocks of lava consumed daily (with buckets returned). Is that an okay spot to have it for longevity, or should it drop a bit to the 12/8/4 I mentioned above?

ProgrammerDan commented:
tbh I have no idea; they don't stack, so that's the real issue. Compactor won't help with trade. I'd say drop it to 4.

MonsieurWTF commented:
Kk, I'll reduce that back too.

MonsieurWTF commented:
Just to summarize so I don't forget what needs to be adjusted:

Wood Crates:
- Change all 384 Dirt -> 192 Gravel

Emerald Crate:
- Change 16 Lava Bucket -> 4 Lava Bucket

MonsieurWTF commented:
@ProgrammerDan By the way quick question, I was thinking on working on something like this in the future: https://docs.google.com/document/d/1BLdQyiNlvW0Gjm7ceL4J8YL31Bf_tQMQh1XIY58rDHo/edit?usp=sharing

Thoughts? (I'll make a thread later as well.)

ProgrammerDan commented:
Wow, I mean, I'd love to see mature technology trees that actually mattered beyond Minecraft's limited enchantment theme; quite an advanced use of FactoryMod. We'd have to run her through the paces to make sure the design worked :D

MonsieurWTF commented:
Yup. The main bad thing that comes to mind would be the config file size climbing significantly per factory that was 'enhance-able'. If I had more capability to code I'd be more willing to experiment beyond the config, but for now that's what I could see as do-able (along with the gruntwork of making every single # of upgrade per tool).

ProgrammerDan commented:
I think it just points to the ultimate unsuitability of FactoryMod for these purposes.

MonsieurWTF commented:
Honestly, it just feels like if FactoryMod had more parameters to track in the config, it'd be a lot more suitable to experimentation beyond its present limited uses. The recipe_output, upgradeable, and use_once parameters are extremely lacking as it now stands, and could be used more efficiently, but I think without the ability to also have recipe_input to get rid of other existing recipes in a factory, and a way to activate improved_recipe after X runs of old_recipe, this is as good as FM gets.

I'll have a thread written up for later, too, discussing the above.

ProgrammerDan commented:
Alright. I'd just generally recommend targetting/getting involved in contraptions if you want to expose this kind of functionality.

MonsieurWTF commented:
Closed for further devleopment.
